### PR TITLE
#44 Email and in-app notifications

### PR DESCRIPTION
**Pull request by patsicko**
# What this PR do
- Create email and in-app notifications
## Description of the  task to be completed

- To send email notifications  to the product lifecycle changes
- To send app notifications to the product lifecycle changes 

## How should this be manually tested?

- git clone from repo
- switch into branch
- run npm i
- ## How should this be manually tested
- clone the repo
- use npm install to install all packages
- Add all required environment variables following the .env.example provided
- use npx sequelize db:migrate to migrate migrations
- Run npm run db:seeds to populate seeders in database
- use npm run dev to start the server
- Navigate to  the http://localhost:PORT/docs on your browser to access swagger APIs docs  by replacing 'PORT' with the port that your the app is listening to. 
- run npm run dev
- try to add product
- check your email  and app for notification:
- Trello ID: #44

### CODE REVIEW

**Comment by patsicko** [comment]
[Notifications](https://trello.com/c/PFoM0IkO/44-notifications)

File: src/controllers/notificationController.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+const notification = await Notification.create({
+  subject:"product deleted",
+  message:`Hello ${vendor.firstName}, a product ${productName} is deleted successfuly`,
+  type:"productDeleted",
+  userId:vendor.id
+  
+  });
+
+  const sendEmaiOption = {
```
Comment (wakaflorien) [inline src/controllers/notificationController.js]: - change the variable name to make it readable and self-explanatory

File: src/controllers/notificationController.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  const notification = await Notification.create({
+    subject:"product sold",
+    message:`Hello ${vendor.firstName}, a product ${productName} is sold from your collection`,
+    type:"productSold",
+    userId:vendor.id
+    
+    });
+
+
+    const sendEmaiOption = {
```
Comment (wakaflorien) [inline src/controllers/notificationController.js]: - same here, variable naming is crucial

**Review by wakaflorien — CHANGES_REQUESTED** [review]

**Review by oliviertech212 — APPROVED** [review]

**Review by chrill-christella — APPROVED** [review]

**Review by Luciefifi — APPROVED** [review]

File: src/controllers/notificationController.js
Comment (sostene12) [inline src/controllers/notificationController.js:1]: you've done a great job @patsicko , you can also make it better by adding more details
- organize the email details in the email templates
- add other endpoints for getting the notifications

**Review by sostene12 — CHANGES_REQUESTED** [review]

**Review by wakaflorien — APPROVED** [review]

**Review by sostene12 — APPROVED** [review]

**Pull request merged**